[IMP] Move the constraint on uniqueness for cutoff date at line level using a configuration

lmarion-source opened the pull request:
+ Add the possibility to create several cutoff with the same date

If the box is not checked in configurations,  the uniqueness is checked at the cutoff level (as previously done). If we need to create several cutoff at the same date for different journals/accounts, we can check the box and the check will be done at the cutoff line level.

Business case can be :

- One user (A) creates a cutoff for some account moves. Another user (B) creates other moves, then wants to see those moves in a cutoff.
  However, B does not want to interfere with the cutoff A already made. B creates a new cutoff with the same date as the one of A, and when he generates lines,
  only retrieves the one that were not already in the cutoff of A.

- One user wants to create several cutoff at the same date, but for different journals/accounts

lmarion-source commented:
@jbaudoux I was not able to re-open the #310 so I did a new one and add a configuration for the business case I have. This way the original behaviour stays the same

acsonefho reviewed (approved):
Code LGTM

jbaudoux commented on account_cutoff_base/migrations/16.0.1.5.1/post-migrate.py:
```diff
@@ -0,0 +1,19 @@
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
```
there is no version change in the manifest for this script

jbaudoux commented on account_cutoff_base/models/account_cutoff_line.py:
```diff
@@ -83,3 +86,54 @@ class AccountCutoffLine(models.Model):
+                )
+        else:
+            read_group = self.env["account.cutoff"].read_group(
+                domain=[("id", "in", self.mapped("origin_move_line_id").ids)],
```
there was no check on origin_move_line_id before. You don't limit to self here like in the other if?

jbaudoux reviewed (changes requested):
you need to make one commit per module

lmarion-source committed:
[IMP] account_cutoff_base: Move the constraint on uniqueness for cutoff date at line level + Add the possibility to create several cutoff with the same date +  Add a configuration to check cutoff date unicity either on lines or on the whole cutoff

Business case can be :

- One user (A) creates a cutoff for some account moves. Another user (B) creates other moves, then wants to see those moves in a cutoff.
      However, B does not want to interfere with the cutoff A already made. B creates a new cutoff with the same date as the one of A, and when he generates lines,
      only retrieves the one that were not already in the cutoff of A.

- One user wants to create several cutoff at the same date, but for different journals/accounts

lmarion-source committed:
[IMP] account_cutoff_start_end_dates: Add a configuration to check cutoff date unicity either on lines or on the whole cutoff
